### PR TITLE
Fix: MonthlyEquivalentCalculator bug

### PR DESCRIPTION
**Pull request by colinbruce**
## What

The EFE team have identified an edge case where the frequency changing causes the calculator to return
incorrect results

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/concerns/monthly_equivalent_calculator.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@ def calculate_monthly_equivalent!(target_field:, collection:, date_method: :paym
 
   def calculate_monthly_equivalent(collection:, date_method: :payment_date, amount_method: :amount)
     @converter = nil # reset the converter each time it's used
+    @frequency = nil
```
Comment (cpjmcquillan) [inline app/models/concerns/monthly_equivalent_calculator.rb:20]: Should we be memoizing `converter` and `frequency` at all?

**Review by cpjmcquillan — APPROVED** [review]
I added a comment for us to think about.

Caching is hard, and a frequent source of bugs generally. If we weren't caching the values, would this bug be possible?

File: app/models/concerns/monthly_equivalent_calculator.rb (hunk 1 above)
Comment (colinbruce) [reply app/models/concerns/monthly_equivalent_calculator.rb:20]: That can be one for the Eligibility Platform team to consider! 😄

Comment (starswan) [reply app/models/concerns/monthly_equivalent_calculator.rb:20]: Converter has to be memoized due to the way the implementation is structured (the memoization is called once to do it, and again for the answer). The structure of this code in CFE-Civil has changed so that local variables are used - which is how we found the bug. We had an new test that accidentally relied on the bug, which then broke when we accidentally fixed it
